app/models/services/crossref.rb

```diff
@@ -0,0 +1,17 @@
+class Crossref
+  def initialize(options={})
+    @base_url = 'http://api.crossref.org/'
+    @metadata_url = @base_url + "works/"
+    @funder_url = @base_url + "funders/"
+  end
+
+  def get_doi_metadata(doi)
+    metadata_uri = URI.parse(@metadata_url+doi)
+    begin
+      JSON.parse Net::HTTP.get(metadata_uri)
+    rescue JSON::ParserError => e
+      return false
```

Comment:
What does returning `false` in the context of `get_doi_metadata` mean, though? A common convention in Ruby is that when a return value is supposed to be an object, it either raises an error or returns `nil`. This way you can either crash quickly or return an object that means, essentially, the object in question wasn't found. That's how the ActiveRecord `#find_by` method works.